photon-core/src/main/java/org/photonvision/vision/videoStream/SocketVideoStream.java

```diff
@@ -53,46 +44,55 @@ public SocketVideoStream(int portID) {
     @Override
     public void accept(Frame frame) {
         if (userCount > 0) {
-            if (jpegBytesLock
-                    .tryLock()) { // we assume frames are coming in frequently. Just skip this frame if we're
-                // locked doing something else.
-                try {
-                    // Does a single-shot frame recieve and convert to JPEG for efficency
-                    // Will not capture/convert again until convertNextFrame() is called
-                    if (frame != null && !frame.image.getMat().empty() && jpegBytes == null) {
-                        frameWasConsumed = false;
-                        jpegBytes = new MatOfByte();
-                        Imgcodecs.imencode(
-                                ".jpg",
-                                frame.image.getMat(),
-                                jpegBytes,
-                                new MatOfInt(Imgcodecs.IMWRITE_JPEG_QUALITY, 75));
-                    }
-                } finally {
-                    jpegBytesLock.unlock();
-                }
+            // if we have a valid frame and at least one user, put it in the queue
+            if (frame != null && !frame.image.getMat().empty()) {
+                var inFrameCopy = new Frame();
+                frame.copyTo(inFrameCopy);
```

Comment:
I don't see a way to avoid this, but it's probably not ideal. Does cscore also make a copy internally, if you know?

Reply:
I don't know for sure. I'm willing to bet no....

Reply:
![image](https://user-images.githubusercontent.com/29715865/209052715-617d1f13-d3a9-4728-8620-f81c22833e22.png)
sure looks like a copy to me